src/vector.rs

```diff
@@ -59,6 +59,14 @@ where
         differences.map(|n| n * n).sum::<f32>().sqrt()
     }
 
+    /// Compute the squared magnitude of a vector
+    fn magnitude_sq(self) -> C
+    where
+        C: core::iter::Sum,
+    {
+        self.iter().map(|n| n * n).sum()
+    }
+
     /// Compute the magnitude of a vector
     fn magnitude(self) -> f32
```

Comment:
It might make sense for `magnitude`  to be composed in terms of `magnitude_sq`. This change also seems irrelevant to the addition of `normalized` and could possibly use its own PR. We might also consider changing `magnitude` to similarly return `C`, and possibly other methods as well, though that seems like a breaking change.

Reply:
I had the same thoughts, just didn't want to fiddle with the existing implementations too much. Expressing `magnitude` in terms of `magnitude_sq` requires the `core::iter::Sum` trait bound as well.

I agree that `magnitude_sq` can be removed here as it doesn't add to `normalize`. I'll pull it into a separate PR.

Reply:
Added in #118.

Reply:
Went a different way now and added the `map` function that transforms elements individually. Let me know what you think about it; if it doesn't fit, I'll remove it.